testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayViewModel.kt

```diff
@@ -52,7 +64,68 @@ class MiniAppDisplayViewModel constructor(
         this.hostLifeCycle = lifecycle
     }
 
-    fun canGoBackwards() : Boolean =
+    fun promptCustomPermission(
```

Comment:
Should not implement UI generating codes in ViewModel. You can also add different ViewModel to interact with `MiniAppDisplayActivity`.

Reply:
Thanks. Currently, I moved the code to a presenter.